timApp/static/scripts/tim/messaging/tim-message-send.component.ts

```diff
@@ -305,10 +305,21 @@ export class TimMessageSendComponent {
         this.timMessageOptions.pageList = "";
     }
 
-    somethingChanged() {
+    somethingChanged(event: Event) {
         this.formChanged = true;
+
+        /* Attach or detach beforeunload listeners, depending on message state */
+        if (this.messageBody !== "" && !this.messageSentOk) {
+            window.addEventListener("beforeunload", this.confirmExitHandler);
+        } else {
+            window.removeEventListener("beforeunload", this.confirmExitHandler);
```

Comment:
Jos lähetän onnistuneesti viestin ja sitten yritän poistua, niin tulee varoitus. Nyt tuo event poistetaan vain jos manuaalisesti tyhjentää tekstin sisällön. Pitäiskö sen olla niin päin että poistuessa tarkistetaan onko lähetys ollut ok ilman uusia muutoksia?

Reply:
Joo, enpä aikaisemmin huomannut, että viestin lähettämisen yhteydessä formia resetoidaan, mukaanlukien tuo `messageSentOk`-muuttujan arvo (jolloin se on tässä kohti aina false).
Lisäsin nyt tähän uuden muuttujan seuraamaan viestin sisällkön muutosta, ja siirsin käsittelijän poiston lähettämisen yhteyteen -> pitäisi nyt toimia oikein.